docs/docs/Concepts/mcp-server.mdx

```diff
@@ -186,31 +186,63 @@ For more information, see the MCP documentation for your client, such as [Cursor
 
 ### MCP server authentication and environment variables {#authentication}
 
-You must provide a Langflow API key in your MCP client configuration if your [Langflow server has authentication enabled](/api-keys-and-authentication#start-a-langflow-server-with-authentication-enabled).
+Each [Langflow project](/concepts-flows#projects) has its own MCP server with its own MCP server authentication settings.
 
-When this is the case, the code template in your project's **MCP Server** tab automatically includes the `--header` and `x-api-key` arguments:
+To configure authentication for a Langflow MCP server, go to the **Projects** page in Langflow, click the **MCP Server** tab, click <Icon name="Fingerprint" aria-hidden="true"/> **Edit Auth**, and then select your preferred authentication method:
 
-```json
-{
-  "mcpServers": {
-    "PROJECT_NAME": {
-      "command": "uvx",
-      "args": [
-        "mcp-proxy",
-        "--headers",
-        "x-api-key",
-        "YOUR_API_KEY",
-        "http://LANGFLOW_SERVER_ADDRESS/api/v1/mcp/project/PROJECT_ID/sse"
-      ]
-    }
-  }
-}
-```
+<Tabs groupId="auth-type">
+<TabItem value="API key" label="API key">
+
+When authenticating your MCP server with a Langflow API key, your project's MCP server **JSON** code snippets and **Auto install** configuration automatically include the `--headers` and `x-api-key` arguments.
 
 Click <Icon name="key" aria-hidden="true"/> **Generate API key** to automatically insert a new Langflow API key into the code template.
 Alternatively, you can replace `YOUR_API_KEY` with an existing Langflow API key.
 
-![MCP server tab showing Generate API key button](/img/mcp-server-api-key.png)
+</TabItem>
+<TabItem value="OAuth" label="OAuth">
+
+When OAuth is enabled, Langflow automatically starts an [MCP Composer](https://pypi.org/project/mcp-composer) instance for your project, creating a secure proxy between MCP clients and the `mcp-proxy` in front of your server.
+
+To configure OAuth authentication:
+
+1. Select **OAuth** as the authentication type.
+2. Configure the OAuth settings with the values from your OAuth deployment:
+
+   - **Host**: OAuth server host
+   - **Port**: OAuth server port
+   - **Server URL**: Full OAuth server URL
+   - **Callback Path**: OAuth callback URL
+   - **Client ID**: Your OAuth client identifier
+   - **Client Secret**: Your OAuth client secret
+   - **Authorization URL**: OAuth authorization endpoint
+   - **Token URL**: OAuth token endpoint
+   - **MCP Scope**: Scope for MCP operations
+   - **Provider Scope**: OAuth provider scope
```

Comment:
Where do these values come from? What do I need to have done beforehand in order to have all of these values to hand? Which are required?

Reply:
Thank you!